### DISCUSSION TITLE
SEQ for multiple environments

### DISCUSSION THREAD

**Discussion by valentyn-bulakh** [datalust/seq-tickets#1230] [Help]
Hi.
Currently we have seq license per environment, but we have plans to  spin up more and we trying to find best approach with logging.
It possible to setup SEQ for multiple environments?
If yes, 
how to separate env's inside one SEQ instance?
what requirements we should consider for the infra? 
maybe you have some recommendation how to setup it in proper way?

**Comment by nblumhardt** [comment]
Hi Valentyn,

Thanks for getting in touch.

The usual formula for supporting multiple environments with one Seq instance is:

 * Apps use different API keys in each environment
 * The API keys have properties applied to them that are attached to incoming events - e.g. a dev API key will carry `Environment = Dev`, in test they'll have `Environment = Test`, etc.
 * Shared signals are then created in the Events screen, one signal per environment, with filters `Environment = 'Dev'`, `Environment = 'Test'`, etc.

You can read a bit more on API keys here: https://docs.datalust.co/docs/api-keys, and signals in: https://docs.datalust.co/docs/signals.

Let me know if this helps, and if you need any further info!

Best regards,
Nick

**Reply by nblumhardt** [reply]
I should add one more thought - it's often worth keeping a separate Seq instance for production vs. the other environments, since runaway logging is more common in dev and test, and it's nice to keep production insulated from these kinds of issues. HTH!

**Comment by valentyn-bulakh** [comment]
Nicholas, 
thank you for the response.
we will try to configure in way that you proposed.

**Reply by nblumhardt** [reply]
That's great - thanks Valentyn.
